analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -126,6 +131,22 @@ kind switch
             _ => null
         };
 
+    private static SymbolicConstraint BinaryNumberConstraint(BinaryOperatorKind kind, NumberConstraint left, NumberConstraint right) =>
+        kind switch
+        {
+            BinaryOperatorKind.Equals when left.IsSingleValue && right.IsSingleValue => BoolConstraint.From(left.Equals(right)),
+            BinaryOperatorKind.NotEquals when left.IsSingleValue && right.IsSingleValue => BoolConstraint.From(!left.Equals(right)),
```

Comment:
Here, we will also need this, but there's no way to cover it yet. A note is added to 
```
BinaryOperatorKind.NotEquals when left.Min > right.Max || left.Max < right.Min => BoolConstraint.False,
```

Reply:
I prefer to add commit that will also add test, to make sure it's correct. (honestly, I made a mistake when writing it on 1st try and edited the message :D )